src/parameters/mod.ts

```diff
@@ -21,46 +37,54 @@ export type PossibleParameterKeys<
 > = (keyof ParameterSetInternal)[];
 
 export type ParameterPropertiesDefinition<
-  Parameters extends ParameterSetDefinition,
-  Required extends PossibleParameterKeys<Parameters>,
+  Params extends ParameterSetDefinition,
+  Required extends PossibleParameterKeys<Params>,
 > = {
-  properties: Parameters;
+  properties: Params;
   required: Required;
 };
 
 export type ParameterVariableType<
   Def extends ParameterDefinition,
-> = Def["type"] extends ICustomType // If the ParameterVariable is a Custom type, use it's definition instead
-  ? ParameterVariableType<Def["type"]["definition"]>
-  : Def extends TypedObjectParameterDefinition // If the ParameterVariable is of type object, allow access to the object's properties
-    ? ObjectParameterVariableType<Def>
+  CurrentDepth extends RecursionDepthLevel = 0,
```

Comment:
It's a complete mystery to me why previously, with no recursion guard whatsoever, the old code didn't lead to an infinite-recursion TS error.